enn_zoo/enn_zoo/griddly_env/level_generators/crafter_generator.py

```diff
@@ -0,0 +1,179 @@
+import functools
+from typing import Any, Dict, List, Set, Tuple, Union
+
+import numpy as np
+import opensimplex
+
+from enn_zoo.griddly_env.level_generators.level_generator import LevelGenerator
+
+
+class CrafterLevelGenerator(LevelGenerator):
+    """
+    Pretty much this: https://github.com/danijar/crafter/blob/main/crafter/worldgen.py
+    """
+
+    def __init__(self, seed: int, width: int, height: int, num_players: int) -> None:
+        self._width = width
+        self._height = height
+        self._num_players = num_players
+        self._walkable = {"G", "P", "S"}
+        self._random = np.random.RandomState(seed)
+
+    def _get_level_string(self, world: np.ndarray) -> str:
+        level_string = ""
+        for h in range(0, self._height):
+            for w in range(0, self._width):
+                level_string += world[w, h].ljust(4)
+            level_string += "\n"
```

Comment:
this is O(n^2), faster way is to create a list with all the string firsts and then use `"".join`